lib/Interpreter/InterOp.cpp

```diff
@@ -2165,5 +2165,46 @@ namespace InterOp {
 
     return names;
   }
+
+  /*
+  FIXME:
+
+  For module GetFunctionArgDefault(TCppFunction_t func, TCppIndex_t param_index)
+  We are specially using a seperate branch for getting default value of floating point variable.
+  The main reason to do this is due to loss of precision obtained due some deficiency in clang.
+  Clang should print the default argument as written and not as represented.
+
+  For example,
+  default argument float PI = 3.14 will be stored as 3.1400000000000001 in variable PI thus causing
+  loss in precision and sometimes generating wrong/invalid output.
+
+  Here's a link for better understanding of problem https://godbolt.org/z/vzWxKdsnh
+  */
+  std::string GetFunctionArgDefault(TCppFunction_t func, TCppIndex_t param_index)
+  {
+    auto *D = (clang::Decl *)func;
+    auto *FD = llvm::dyn_cast_or_null<clang::FunctionDecl>(D);
+    auto PI = FD->getParamDecl(param_index);
+
+    if (PI->hasDefaultArg())
+    {
+      std::string Result;
+      llvm::raw_string_ostream OS(Result);
+      Expr *DefaultArgExpr = const_cast<Expr *>(PI->getDefaultArg());
+      DefaultArgExpr->printPretty(OS, nullptr, PrintingPolicy(LangOptions()));
+
+      if (PI->getType()->isFloatingType())
```

Comment:
```suggestion
      // FIXME: Floats printed in clang with the precision of their underlying representation
      // and not as written. This is a deficiency in the printing mechanism of clang which we
      // extra work to mitigate. For example float PI = 3.14 is printed as 3.1400000000000001.
      if (PI->getType()->isFloatingType())
```

Can you delete the fixme above and use this one instead?

Reply:
I have made changes in FIXME @vgvassilev as mentioned by you.
Are there any further changes/improvements required pls let me know so that I could start working on it

Reply:
Again, single commit message which reads well.

Reply:
- I hope its according to your requirements